src/components/monasca_client.js

```diff
@@ -235,59 +234,62 @@ export default class MonascaClient {
     }
 
     return this.backendSrv.get("api/plugins/monasca-app/settings")
-      .then(response => {
-	if (!response.jsonData || !response.jsonData.datasourceName) {
-	  throw { message: 'No datasource selected in app configuration' };
-	}
-	return this.datasourceSrv.get(response.jsonData.datasourceName)
-	  .then(ds => {
-	    this.ds = ds;
-	    return this.ds;
-	  })
-	  .catch(err => { throw err; });
-      })
-      .catch(err => { throw err; });
+    .then(response => {
+	    if (!response.jsonData || !response.jsonData.datasourceName) {
+	      throw { message: 'No datasource selected in app configuration' };
+	    }
+      
+      return this.datasourceSrv.get(response.jsonData.datasourceName)
+	            .then(ds => {
+                console.log(ds);
+	              this.ds = ds;
+                return this.ds;
+	            })
+	            .catch(err => { throw err; });
+    })
+    .catch(err => { throw err; });
   }
 
   _request(method, path, params, data) {
-    return this._getDataSource().then(data_source => {
-
-      var headers = {
-	'Content-Type': 'application/json',
-	'X-Auth-Token': data_source.token
-      };
-
-      var options = {
-	method: method,
-	url:    data_source.url + path,
-	params: params,
-	data: data,
-	headers: headers,
-	withCredentials: true,
-      };
-
-      return data_source.backendSrv.datasourceRequest(options).catch(err => {
-	if (err.status !== 0 || err.status >= 300) {
-          var monasca_response;
-          if (err.data) {
-            if (err.data.message){
-              monasca_response = err.data.message;
-	    }
-	    else if (err.data.description) {
-	      monasca_response = err.data.description;
-	    } else{
-              var err_name = Object.keys(err.data)[0];
-              monasca_response = err.data[err_name].message;
-            }
-          }
-          if (monasca_response) {
-            throw { message: 'Monasca Error Response: ' + monasca_response };
-          } else {
-            throw { message: 'Monasca Error Status: ' + err.status };
-          }
-	}
-      });
-    }).catch(err => { throw err; });
+    return Promise.reject( { message: 'MY ERROR!'} )
+  //   return this._getDataSource().then(data_source => {
```

Comment:
I guess commenting this out isn't supposed to be in the commit?

If you haven't figured it out already (I'm sure you have :) ) you can use ```git rebase -i <commit_id>``` to go back and amend previous commits.